src/Core/Layer/Layer.js

```diff
@@ -53,11 +53,18 @@ export const defineLayerProperty = function defineLayerProperty(layer, propertyN
     }
 };
 
-function GeometryLayer(i) {
+function GeometryLayer(id, object) {
+    if (!id) {
+        throw new Error('Missing id parameter (GeometryLayer must have a unique id defined)');
+    }
+    if (!object || !object.isObject3D) {
+        throw new Error('Missing/Invalid object parameter (must be a three.js Object3D instance)');
```

Comment:
I don't have access to the `scene` here so I prefered to require the `object` param.